3rdparty/find_dependencies.cmake

```diff
@@ -867,29 +869,31 @@ if(USE_SYSTEM_CURL)
         set(USE_SYSTEM_CURL OFF)
     endif()
 endif()
-if(NOT USE_SYSTEM_CURL)
-    if(USE_SYSTEM_OPENSSL)
-        open3d_find_package_3rdparty_library(3rdparty_openssl
-            PACKAGE OpenSSL
-            TARGETS OpenSSL::Crypto
-        )
-        if(NOT 3rdparty_openssl_FOUND)
-            set(USE_SYSTEM_OPENSSL OFF)
-        endif()
-    endif()
-    if(NOT USE_SYSTEM_OPENSSL)
-        # BoringSSL
-        include(${Open3D_3RDPARTY_DIR}/boringssl/boringssl.cmake)
-        open3d_import_3rdparty_library(3rdparty_openssl
-            INCLUDE_DIRS ${BORINGSSL_INCLUDE_DIRS}
-            INCLUDE_ALL
-            INCLUDE_DIRS ${BORINGSSL_INCLUDE_DIRS}
-            LIB_DIR      ${BORINGSSL_LIB_DIR}
-            LIBRARIES    ${BORINGSSL_LIBRARIES}
-            DEPENDS      ext_zlib ext_boringssl
-        )
+
+if(USE_SYSTEM_OPENSSL)
+    open3d_find_package_3rdparty_library(3rdparty_openssl
+        PACKAGE OpenSSL
+        REQUIRED
+        TARGETS OpenSSL::Crypto
+    )
+    if(NOT 3rdparty_openssl_FOUND)
+        set(USE_SYSTEM_OPENSSL OFF)
     endif()
+endif()
+if(NOT USE_SYSTEM_OPENSSL)
+    # BoringSSL
+    include(${Open3D_3RDPARTY_DIR}/boringssl/boringssl.cmake)
+    open3d_import_3rdparty_library(3rdparty_openssl
+        INCLUDE_DIRS ${BORINGSSL_INCLUDE_DIRS}
+        INCLUDE_ALL
+        INCLUDE_DIRS ${BORINGSSL_INCLUDE_DIRS}
+        LIB_DIR      ${BORINGSSL_LIB_DIR}
+        LIBRARIES    ${BORINGSSL_LIBRARIES}
+        DEPENDS      ext_zlib ext_boringssl
+    )
+endif()
 
+if(NOT USE_SYSTEM_CURL)
```

Comment:
Hi @christian-rauch we need SSL as an indirect dependency of cURL. So with USE_SYSTEM_CURL, we do not expliclty need to find and link system SSL. Can you update this so that SSL is ignored with USE_SYSTEM_CURL?

Reply:
But isn't this exactly what the commit is fixing? Previously, with `USE_SYSTEM_CURL` set to `ON`, we would not link SSL. Hence, the linking issues. So with `USE_SYSTEM_CURL=ON`, we still need to link SSL explicitly.

Reply:
@ssheorey